plot/plot.app.js

```diff
@@ -1157,16 +1289,128 @@
+                                                                }
+                                                                label: //label on which click event took place
+                                                                text:""
+                                                            }
+                                                        }
+                                                    }
+                                                    // For bar chart
+                                                
+                                                    data:{
+                                                        event:{..}
+                                                        points:{
+                                                            0:{
+                                                                data:{
+                                                                    x:[ //array of labels in bar chart xaxis],
+                                                                    y: [ //array of labels in bar chart yaxis ]
+                                                                    graphic_link_pattern: // array of links which are used to make chart clickable
+                                                                },
+                                                                fullData:{
+                                                                    // Data with default parameters used by plotly
+                                                                    alignmentgroup:"",
+                                                                    cliponxaxis: true,
+                                                                    x:[],
+                                                                    y:[]
+                                                                }
+                                                                label: //label on which click event took place
+                                                                text:""
+                                                            }
+                                                        }
+                                                    }
+                                                    
+                                                    
+                                                    */
+
+                                                    if(data.hasOwnProperty("points") && data.points[0].data.hasOwnProperty("graphic_link_pattern")){
+                                                        var idx="";
+                                                        switch(data.points[0].data.type){
+                                                            case "pie":
+                                                                idx=data.points[0].data.labels.indexOf(data.points[0].label.toString());
```

Comment:
Could you add a comment about what `data.points[0].label` is here and why you need the idx of this label based on the list of `data.points[0].data.labels`.

My understanding is that this is the label for the current slice which is in an array of labels. And that this array of labels should be the same length as the `data.points[0].data.graphic_link_pattern` array used below:
```
labels = ["label1", "label2", ...]
graphic_link_pattern = ["www.google.com", "www.google.com", ...]
```

Reply:
Added the comment